src/simulation/power/simpleBattery/simpleBattery.h

```diff
@@ -39,6 +39,8 @@ class SimpleBattery: public PowerStorageBase {
 
 public:
     double storageCapacity; //!< [W-s] Battery capacity in Watt-seconds (Joules).
+    bool fault = false; //!< Fault flag
```

Comment:
These variables should be private variables, and then we need associated setter and getter functions

Reply:
If you’re using setters, then i would move the check `0<=faultCapacityRatio<=1` to the setter of `faultCapacityRatio` instead of every time the model is run.

Reply:
But, if we are reading in the value from a message, then we don't have that control. I like the idea of moving the fault info to a battery fault message.  I think it will provide a more elegant solution.